lib/fauna_model/base.rb

```diff
@@ -0,0 +1,269 @@
+module Fauna
+  module Model
+    class Base
+      include ActiveModel::Validations
+      include Fauna::Model::Dirty
+
+      class << self
+        def fauna_class
+          @fauna_class || fail(InvalidClass.new('No class set'))
+        end
+
+        def fauna_class=(fauna_class)
+          fauna_class = Fauna::Ref.new(fauna_class.to_s) unless fauna_class.is_a? Fauna::Ref
+          @fauna_class = fauna_class
```

Comment:
Regardless of what you do here, this shadowing of the param with a local var is a bit gross.